internal/controllers/addoninstance/addon_instance_controller.go

```diff
@@ -0,0 +1,127 @@
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+const (
+	// Default timeout when we do a manual RequeueAfter
+	defaultRetryAfterTime = 10 * time.Second
+)
+
+// Global Flag for Install State
+var installState int
+
+type StatusControllerReconciler struct {
+	client              client.Client
+	addonInstanceClient *addoninstance.AddonInstanceClientImpl
+}
+
+// Review Grab Namespace at this stage (TO BE FIXED)
+func NewStatusControllerReconciler(client client.Client) *StatusControllerReconciler {
+	ctx, desiredAddonInstance := addoninstancereconcile.ensureReferenceAddon(&StatusControllerReconciler)
+	return &StatusControllerReconciler{
+		client:              client,
+		addonInstanceClient: addoninstance.NewAddonInstanceClient(client),
+		name:                desiredAddonInstance.Name,
+		namespace:           desiredAddonInstance.Namespace,
+	}
+}
+
+// Watch reference addon actions to trigger addon instance
+func (r *StatusControllerReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&rv1alpha1.ReferenceAddon{}).
+		Complete(r)
+}
+
+// Utilize info gathered from SetupWithManager to perform logic against
+func (s *StatusControllerReconciler) Reconcile(ctx context.Context) (ctrl.Result, error) {
+	log := log.Log.WithName("reference addon - addon instance")
+	addonClient := NewStatusControllerReconciler(s.client)
+	key := client.ObjectKey{
+		//Review Put values obtained from constructor here instead
+		Namespace: addonClient.namespace,
+		Name:      addonClient.name,
+	}
+
+	log.Info("find reference addon: ", key.Namespace, key.Name)
+	referenceaddon := &rv1alpha1.ReferenceAddon{}
+	if err := s.client.Get(ctx, key, referenceaddon); err != nil {
+		return ctrl.Result{}, fmt.Errorf("getting Reference-Addon Object'%s/%s': %w", key.Namespace, key.Name, err)
+	}
+
+	log.Info("find addon instance: ", key.Namespace, key.Name)
+	addonInstance := &av1alpha1.AddonInstance{}
+	if err := s.client.Get(ctx, key, addonInstance); err != nil {
+		return ctrl.Result{}, fmt.Errorf("getting AddonInstance Object'%s/%s': %w", key.Namespace, key.Name, err)
+	}
+	log.Info("found addon instance: ", addonInstance)
+
+	getAddonInstanceUpdateCondition(referenceaddon, addonInstance, log)
+
+	var updatedInstance av1alpha1.AddonInstance
+
+	//Condition for install state successful
+	installSuccess := []metav1.Condition{
+		addoninstance.NewAddonInstanceConditionInstalled(
+			"True",
+			av1alpha1.AddonInstanceInstalledReasonSetupComplete,
+			"All components up",
+		),
+	}
+
+	degradedState := []metav1.Condition{
+		addoninstance.NewAddonInstanceConditionDegraded(
+			"True",
+			string(av1alpha1.AddonInstanceConditionDegraded),
+			"Service Degraded",
+		),
+	}
+	// Send Pulse to gather info from installation of addon instance
+	err := s.addonInstanceClient.SendPulse(ctx, *addonInstance, nil)
```

Comment:
`SendPulse` should be the final instruction here and is used for sending status not receiving it.